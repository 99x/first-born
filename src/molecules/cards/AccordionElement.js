import React, { Component } from "react";
import {
    View,
    StyleSheet,
    TouchableOpacity,
    Animated,
    Platform
} from "react-native";
import PropTypes from "prop-types";

import { Text } from "../../atoms/Text";
import { Icon } from "../../atoms/Icon";
import { commonColors } from "../../utils/color";

export class AccordionElement extends Component {
    constructor(props) {
        super(props);
        this.state = {
            expanded: props.expanded
        };
        this.fadeAnim = new Animated.Value(0);
    }

    animate() {
        this.fadeAnim = new Animated.Value(0);
        Animated.timing(this.fadeAnim, {
            toValue: 1,
            duration: 250,
            useNativeDriver: true
        }).start();
    }

    handleShowHide = () => {
        this.setState({ expanded: !this.state.expanded });
        this.animate();
    };

    render() {
        const {
            title,
            description,
            image,
            onPress,
            block,
            bodyColor,
            headerColor,
            style,
            isList,
            ...otherProps
        } = this.props;

        const { expanded } = this.state;

        let cardStyle = [
            styles.container,
            { borderColor: headerColor, borderWidth: 1 }
        ];
        const headerStyle = [
            styles.titleContainer,
            { backgroundColor: headerColor }
        ];
        const bodyStyle = {
            ...styles.bodyContainer,
            backgroundColor: bodyColor
        };

        if (!isList) {
            cardStyle.push({ borderRadius: 5 });
        }

        if (style) {
            cardStyle.push(style);
        }

        return (
            <View style={cardStyle}>
                <TouchableOpacity
                    onPress={this.handleShowHide}
                    style={headerStyle}
                    {...otherProps}
                >
                    <Text
                        color={commonColors.darkGrey}
                        bold
                        style={{ flex: 1 }}
                    >
                        {title}
                    </Text>
                    {this.renderIcon()}
                </TouchableOpacity>
                {expanded && (
                    <Animated.View
                        style={[bodyStyle, { opacity: this.fadeAnim }]}
                    >
                        <Text size="footnote" color={commonColors.darkGrey}>
                            {description}
                        </Text>
                    </Animated.View>
                )}
            </View>
        );
    }

    renderIcon() {
        const { expandedIcon, icon } = this.props;
        const { expanded } = this.state;
        return (
            <Icon
                name={expanded ? expandedIcon : icon}
                size={20}
                color={commonColors.inputGrey}
            />
        );
    }
}

AccordionElement.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    bodyColor: PropTypes.string,
    headerColor: PropTypes.string,
    expanded: PropTypes.bool,
    expandedIcon: PropTypes.string,
    icon: PropTypes.string,
    isList: PropTypes.bool,
    ...TouchableOpacity.propTypes
};

AccordionElement.defaultProps = {
    bodyColor: commonColors.white,
    headerColor: commonColors.lightGrey,
    isList: false,
    expanded: false,
    expandedIcon: Platform.OS === "ios" ? "arrow-up" : "arrow-dropup",
    icon: Platform.OS === "ios" ? "arrow-down" : "arrow-dropdown"
};

const styles = StyleSheet.create({
    container: {
        width: "100%",
        flex: 1,
        flexDirection: "column",
        overflow: "hidden"
    },
    titleContainer: {
        paddingVertical: 5,
        paddingHorizontal: 10,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
        overflow: "hidden"
    },
    bodyContainer: {
        flexDirection: "row",
        flex: 1,
        padding: 20,
        overflow: "hidden"
    }
});
