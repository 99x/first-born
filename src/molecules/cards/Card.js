import React, { Component } from "react";
import { View, StyleSheet, TouchableOpacity, Image, Platform, Dimensions } from "react-native";
import PropTypes from "prop-types";

import { Text } from "../../atoms/Text";
import { commonColors } from "react-native-atom-elements/src/utils/color";

const { width } = Dimensions.get("window");

export class Card extends Component {

    render() {
        const { title, description, image, onPress, block, backgroundColor, style, ...otherProps } = this.props;

        const cardBackColor = backgroundColor ? backgroundColor : commonColors.white;

        let cardStyle = block ? [styles.containerBlock, { backgroundColor: cardBackColor }] : [styles.container, , { backgroundColor: cardBackColor }];

        if (style) {
            cardStyle.push(style);
        }

        return (
            <TouchableOpacity style={cardStyle} disabled={!onPress} onPress={onPress} {...otherProps}>
                {image && <Image source={image} style={styles.image} />}
                <View style={styles.textContainer}>
                    <Text size="h6" color="rgba(33, 33, 33, 0.87)" bold >{title}</Text>
                    {description && <Text size="sub_heading" color="rgba(33, 33, 33, 0.4)">{description}</Text>}
                </View>
            </TouchableOpacity>
        )
    }
}

Card.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    image: PropTypes.any,
    block: PropTypes.bool,
    backgroundColor: PropTypes.string,
    ...TouchableOpacity.propTypes
}

const styles = StyleSheet.create({
    container: {
        width: "100%",
        shadowColor: "#000",
        shadowOffset: { width: 3, height: 3 },
        shadowOpacity: 0.3,
        shadowRadius: 3,
        elevation: 3,
        borderRadius: 3,
        overflow: "hidden",
        marginBottom: 5
    },
    containerBlock: {
        width,
        overflow: "hidden",
        borderTopWidth: 1,
        borderBottomWidth: 1,
        borderTopColor: "rgba(33, 33, 33,  0.4)",
        borderBottomColor: "rgba(33, 33, 33,  0.4)",
        marginBottom: 5
    },
    textContainer: {
        padding: 20,
        flexDirection: 'column',
        justifyContent: 'flex-start',
    },
    image: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        height: 150,
        borderTopLeftRadius: 3,
        borderTopRightRadius: 3
    }
})