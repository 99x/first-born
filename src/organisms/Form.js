import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../atoms/Picker";

import { FormDatePicker } from "../molecules/formElements/FormDatePicker";
import { FormInput } from "../molecules/formElements/FormInput";
import { FormPicker } from "../molecules/formElements/FormPicker";
import { FormTextArea } from "../molecules/formElements/FormTextArea";

export class Form extends Component {
    render() {
        const { color, formElements, ...otherProps } = this.props;

        const inputColor = color ? color : "#486c86";

        return (
            <View style={styles.container}>
                {formElements.map((element, key) => {
                    return (
                        <View style={styles.formElementContainer} key={key}>
                            {this.renderElements(element, inputColor, otherProps)}
                        </View>
                    );
                })}
            </View>
        );
    }

    renderElements(element, color, styleProps) {
        element.color = color;
        const { type, pickerData, ...otherProps } = element;

        switch (type) {
            case "text":
                return <FormInput {...styleProps} {...otherProps} />;
            case "textarea":
                return <FormTextArea {...styleProps} {...otherProps} />;
            case "date":
                return <FormDatePicker {...styleProps} {...otherProps} />;
            case "picker":
                return (
                    <FormPicker {...styleProps} {...otherProps}>
                        {pickerData.map((dataElement, key) => (
                            <Picker.Item {...dataElement} key={key} />
                        ))}
                    </FormPicker>
                );
        }
    }
}

Form.propTypes = {
    color: PropTypes.string,
    activeStyle: PropTypes.object,
    errorStyle: PropTypes.object,
    rounded: PropTypes.bool,
    underline: PropTypes.bool,
    defaultStyle: PropTypes.bool,
    noStyle: PropTypes.bool,
    errorColor: PropTypes.string,
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: "100%"
    },
    formElementContainer: {
        marginTop: 10
    }
});
