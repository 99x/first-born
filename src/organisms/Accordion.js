import React, { Component } from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";
import { AccordionElement } from "../molecules/cards/AccordionElement";

export class Accordion extends Component {
    render() {
        const { horizontal, renderItem, ...otherProps } = this.props;

        return (
            <FlatList
                keyExtractor={this.keyExtractor}
                renderItem={
                    renderItem
                        ? renderItem
                        : ({ item }) => {
                              return (
                                  <AccordionElement
                                      {...item}
                                      isList
                                      {...otherProps}
                                  />
                              );
                          }
                }
                {...otherProps}
            />
        );
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

Accordion.propTypes = {
    bodyColor: PropTypes.string,
    headerColor: PropTypes.string,
    expandedIcon: PropTypes.string,
    icon: PropTypes.string,
    ...FlatList.propTypes
};
