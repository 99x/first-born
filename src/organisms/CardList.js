import React, { Component } from "react";
import { FlatList, Dimensions, View, Platform } from "react-native";
import { Card } from "../molecules/cards/Card";

const { width } = Dimensions.get("window");

export class CardList extends Component {

    render() {
        const { backgroundColor, horizontal, margin, ...otherProps } = this.props;

        return (
            <FlatList
                style={{ marginTop: 5 }}
                keyExtractor={this.keyExtractor}
                renderItem={({ item }) => {
                    return (
                        <View style={{ marginRight: margin }}>
                            <Card image={item.image} title={item.title} description={item.description} backgroundColor={backgroundColor} style={{ width: width - 40 }} />
                        </View>
                    )
                }}
                horizontal
                {...otherProps} />
        )
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

CardList.propTypes = {
    ...FlatList.propTypes
}

CardList.defaultProps = {
    margin: Platform.OS === "ios" ? 15 : 10
}
