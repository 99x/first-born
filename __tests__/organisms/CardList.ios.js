import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { CardList } from "../../src/organisms/CardList";
import { Card } from "../../src/molecules/cards/Card";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "ios";
    return Platform;
});

it("renders a list of cards with heading", () => {
    const listData = [
        { title: "Heading 1" },
        { title: "Heading 2" },
        { title: "Heading 3" }
    ];

    const tree = renderer.create(<CardList data={listData} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of cards with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer.create(<CardList data={listData} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a horizontal list of cards with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(<CardList data={listData} horizontal />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a colored list of cards with heading and title", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(<CardList data={listData} backgroundColor="aliceblue" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a list of cards with heading and title by a user defined renderItem method", () => {
    const listData = [
        { title: "Heading 1", description: "Description 1" },
        { title: "Heading 2", description: "Description 2" },
        { title: "Heading 3", description: "Description 3" }
    ];

    const tree = renderer
        .create(
            <CardList
                data={listData}
                renderItem={({ item }) => <Card {...item} />}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
