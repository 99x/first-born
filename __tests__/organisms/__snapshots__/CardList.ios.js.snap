// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a colored list of cards with heading and title 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Description 1",
        "title": "Heading 1",
      },
      Object {
        "description": "Description 2",
        "title": "Heading 2",
      },
      Object {
        "description": "Description 3",
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 2
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "aliceblue",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 3
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders a horizontal list of cards with heading and title 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Description 1",
        "title": "Heading 1",
      },
      Object {
        "description": "Description 2",
        "title": "Heading 2",
      },
      Object {
        "description": "Description 3",
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 5,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "marginRight": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 710,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "marginRight": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 710,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 2
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "marginRight": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 710,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 3
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders a list of cards with heading 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "title": "Heading 1",
      },
      Object {
        "title": "Heading 2",
      },
      Object {
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders a list of cards with heading and title 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "description": "Description 1",
        "title": "Heading 1",
      },
      Object {
        "description": "Description 2",
        "title": "Heading 2",
      },
      Object {
        "description": "Description 3",
        "title": "Heading 3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 2
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 2
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "margin": 5,
              "opacity": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 5,
              "width": 740,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.87)",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Heading 3
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(33, 33, 33, 0.4)",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Description 3
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
